internal/pkg/ping/ping_hander.go

```diff
@@ -0,0 +1,24 @@
+package ping
```

Comment:
I dislike putting logic into `main.go` because it's _harder_ to test and I _want_ to test `/ping`. 

They _do need_ to be handled by a middleware as high up the middleware stack as possible since they need to respond to _every_ host header. We can't just place them alongside our other routes in the mux setup because those routes _only_ apply to registered upstream `From:` blocks.

Reply:
Gotcha, needing to test that `/ping` will respond to any request hadn't occurred to me.  In that case, how would you feel about changing the name & API of this package to make its use a bit more obvious?

```go
package healthcheck

func Middleware(healthcheckPath string, next http.Handler) http.Handler
```

where we'd specify the `"/ping"` path when applying the middleware, to move that path at least a bit closer to where other routing decisions/information lives?